app/src/test/java/com/futurice/freesound/feature/home/HomeFragmentViewModelTest.java

```diff
@@ -0,0 +1,110 @@
+package com.futurice.freesound.feature.home;
+
+import com.futurice.freesound.feature.common.Navigator;
+import com.futurice.freesound.network.api.model.AvatarResult;
+import com.futurice.freesound.network.api.model.UserResult;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+
+import android.support.annotation.NonNull;
+
+import java.util.concurrent.atomic.AtomicBoolean;
+
+import io.reactivex.Single;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+public class HomeFragmentViewModelTest {
+
+    private static final UserResult USER = createUser();
+
+    @Mock
+    private UserDataModel userDataModel;
+
+    @Mock
+    private Navigator navigator;
+
+    private HomeFragmentViewModel viewModel;
+
+    @Before
+    public void setUp() {
+        MockitoAnnotations.initMocks(this);
+    }
+
+    @Test
+    public void getImage_returnsLargeImage() {
+        when(userDataModel.getHomeUser()).thenReturn(Single.just(USER));
+
+        viewModel = createVm();
+
+        viewModel.getImage()
+                 .test()
+                 .assertValue(USER.avatar().large());
+    }
+
+    @Test
+    public void getUserName_returnsName() {
+        when(userDataModel.getHomeUser()).thenReturn(Single.just(USER));
+        viewModel = createVm();
+
+        viewModel.getUserName()
+                 .test()
+                 .assertValue(USER.username());
+    }
+
+    @Test
+    public void getAbout_returnsAbout() {
+        when(userDataModel.getHomeUser()).thenReturn(Single.just(USER));
+        viewModel = createVm();
+
+        viewModel.getAbout()
+                 .test()
+                 .assertValue(USER.about());
+    }
+
+    @Test
+    public void getHomeUser_doesNotSubscribeInCtor() {
+        AtomicBoolean isSubscribed = new AtomicBoolean();
+        when(userDataModel.getHomeUser())
+                .thenReturn(Single.<UserResult>never()
+                                    .doOnSubscribe(__ -> isSubscribed.set(true)));
+
+        viewModel = createVm();
+
+        assertThat(isSubscribed.get()).isFalse();
+    }
+
+    @Test
+    public void getProperties_subscribesOnlyOnce() {
+        when(userDataModel.getHomeUser()).thenReturn(Single.just(USER));
+        viewModel = createVm();
+
+        viewModel.getImage().test();
+        viewModel.getUserName().test();
+
+        verify(userDataModel, times(1)).getHomeUser();
+    }
+
+    @NonNull
+    private HomeFragmentViewModel createVm() {
+        return new HomeFragmentViewModel(userDataModel, navigator);
+    }
+
+    private static UserResult createUser() {
```

Comment:
I'm not really a big fan of mocking data objects; especially if they have a nice builder interface. Could you use that builder instead to create the test object.